expression/builtin_miscellaneous_vec.go

```diff
@@ -359,13 +359,11 @@ func doSleep(secs float64, sessVars *variable.SessionVars) (isKilled bool) {
 	for {
 		select {
 		case <-ticker.C:
-			// Regular kill
-			if atomic.CompareAndSwapUint32(&sessVars.Killed, 1, 0) {
-				timer.Stop()
-				return true
-			}
-			// Killed because of max execution time
-			if atomic.CompareAndSwapUint32(&sessVars.Killed, 2, 0) {
+			// Regular kill or Killed because of max execution time.
+			if atomic.LoadUint32(&sessVars.Killed) == 1 || atomic.LoadUint32(&sessVars.Killed) == 2 {
+				if len(sessVars.StmtCtx.TableIDs) == 0 {
+					atomic.StoreUint32(&sessVars.Killed, 0)
+				}
```

Comment:
It's ok to seperate compare-and-swap into load and store when `len(sessVars.StmtCtx.TableIDs) == 0`, the `sessVars.Killed` atomic variable is used by sleep thread only. But can we make the code like this because I think load and store in two places are strange when doing atomic operation:

```go
if  len(sessVars.StmtCtx.TableIDs) == 0 {
	// the compare-and-swap code before
} else {
	if atomic.LoadUint32(&sessVars.Killed) == 1 || atomic.LoadUint32(&sessVars.Killed) == 2 {
		return true
	}
}